react/src/pages/utils/components/GridFieldsDisplay.tsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { Grid, GridProps } from "@material-ui/core";
+import { Field } from "../typings";
+import FieldDisplayEditable from "./FieldDisplayEditable";
+
+// Grids are very picky about what sizes are allowed
+type Width = Exclude<GridProps["xs"], boolean | "auto" | undefined>;
```

Comment:
Does `GridProps["xs"]` not work even though that's the defined type?

Reply:
I meant to specify that only integers between 1 and 12 are allowed. I originally wrote `1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12` for the type but decided to change it to this instead.